packages/xsnap/src/xsnap.js

```diff
@@ -98,10 +107,28 @@ export function xsnap(options) {
 
   assert(!/^-/.test(name), `name '${name}' cannot start with hyphen`);
 
+  /** @type {(() => Promise<void>) | undefined} */
+  let startCleanup;
+
   let args = [name];
-  if (snapshot) {
-    args.push('-r', snapshot);
-  }
+  await (snapshot &&
+    (async () => {
+      const { hint = 'unknown' } = /** @type {any} */ (snapshot);
```

Comment:
huh, the argument is a stream, but it might have a `.hint` property? is that common?

Reply:
Yeah poor attempt at plumbing through a description for the tracing feature. Will redo.